### Why Handle Missing Values?
#### Missing values can arise due to data collection errors, system failures, or incomplete responses. If left unaddressed, they can lead to biased conclusions and poor model performance.

### Methods for Handling Missing Values
### There are several ways to handle missing values:

### Removing Data
### Removing columns with too many missing values.
### Removing rows that contain missing values.
### Filling Missing Values (Imputation)
#### For numerical data: Use mean, median, or mode.
#### For categorical data: Use mode or introduce a new category.
#### Advanced Imputation Techniques
#### Interpolation to estimate missing values.
#### Predictive models to estimate missing values based on other variables.

In [33]:
import pandas as pd

In [34]:
import numpy as np


In [35]:
data=pd.read_csv("googleplaystore.csv")

## handiling missing values

In [36]:
data.head()
#df.head(10)

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [37]:
data.isnull().sum()


App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [38]:
df=data.dropna()
#pd.read_csv("googleplaystore.csv").dropna()

# Numeric data
### similarly do the same for every column in dataset

In [39]:
count = 0
for i in df['Rating']:
    count+=i
print(count)

39235.60000000005


In [40]:
df['Rating']

0        4.1
1        3.9
2        4.7
3        4.5
4        4.3
        ... 
10834    4.0
10836    4.5
10837    5.0
10839    4.5
10840    4.5
Name: Rating, Length: 9360, dtype: float64

In [42]:
int(sum(df['Rating']))

39235

In [41]:
sum(df['Rating'])

39235.6

### for average rating we need given sum  by total

In [44]:
avg_rating = int(sum(df['Rating']))/9360
print(avg_rating)

4.191773504273504


# "Categorie Data"
#### similarly do for every category data column

In [45]:
df.Category

0             ART_AND_DESIGN
1             ART_AND_DESIGN
2             ART_AND_DESIGN
3             ART_AND_DESIGN
4             ART_AND_DESIGN
                ...         
10834                 FAMILY
10836                 FAMILY
10837                 FAMILY
10839    BOOKS_AND_REFERENCE
10840              LIFESTYLE
Name: Category, Length: 9360, dtype: object

In [49]:
df['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

### to know the count of each categorytype in category column

In [55]:
count = 0
for i in df['Category']:
    if(i=='ART_AND_DESIGN'):
        count+=1
print('ART_AND_DESIGN',count)

ART_AND_DESIGN 61


In [56]:
count = 0
for i in df['Category']:
    if(i=='BUSINESS'):
        count+=1
print('BUSINESS',count)

BUSINESS 303


### to avoid manual count for each category @2 methods
#### -run for loop for each categorytype to the given column
#### *create a dictory to store each caterogy count
#### *run for loop for each unique category
#### *count the items for each categorytype
#### store in dictonary

In [61]:
categories = {}
for name in df['Category'].unique():
    count = 0
    for i in df['Category']:
        if(i == name):
            count+=1
        categories[name] = count
            
print(categories)
            

{'ART_AND_DESIGN': 61, 'AUTO_AND_VEHICLES': 73, 'BEAUTY': 42, 'BOOKS_AND_REFERENCE': 178, 'BUSINESS': 303, 'COMICS': 58, 'COMMUNICATION': 328, 'DATING': 195, 'EDUCATION': 155, 'ENTERTAINMENT': 149, 'EVENTS': 45, 'FINANCE': 323, 'FOOD_AND_DRINK': 109, 'HEALTH_AND_FITNESS': 297, 'HOUSE_AND_HOME': 76, 'LIBRARIES_AND_DEMO': 64, 'LIFESTYLE': 314, 'GAME': 1097, 'FAMILY': 1746, 'MEDICAL': 350, 'SOCIAL': 259, 'SHOPPING': 238, 'PHOTOGRAPHY': 317, 'SPORTS': 319, 'TRAVEL_AND_LOCAL': 226, 'TOOLS': 733, 'PERSONALIZATION': 312, 'PRODUCTIVITY': 351, 'PARENTING': 50, 'WEATHER': 75, 'VIDEO_PLAYERS': 160, 'NEWS_AND_MAGAZINES': 233, 'MAPS_AND_NAVIGATION': 124}


In [62]:
for name in df['Category'].unique():
    count = 0
    for i in df['Category']:
        if(i == name):
            count+=1
    print(name,":",count)


ART_AND_DESIGN : 61
AUTO_AND_VEHICLES : 73
BEAUTY : 42
BOOKS_AND_REFERENCE : 178
BUSINESS : 303
COMICS : 58
COMMUNICATION : 328
DATING : 195
EDUCATION : 155
ENTERTAINMENT : 149
EVENTS : 45
FINANCE : 323
FOOD_AND_DRINK : 109
HEALTH_AND_FITNESS : 297
HOUSE_AND_HOME : 76
LIBRARIES_AND_DEMO : 64
LIFESTYLE : 314
GAME : 1097
FAMILY : 1746
MEDICAL : 350
SOCIAL : 259
SHOPPING : 238
PHOTOGRAPHY : 317
SPORTS : 319
TRAVEL_AND_LOCAL : 226
TOOLS : 733
PERSONALIZATION : 312
PRODUCTIVITY : 351
PARENTING : 50
WEATHER : 75
VIDEO_PLAYERS : 160
NEWS_AND_MAGAZINES : 233
MAPS_AND_NAVIGATION : 124


### shortcut to know the no of categories and popular category in column
#### count       9360 ---> total rows in column category
#### unique        33----> no of categories in column
#### top       FAMILY-------> popular category
#### freq        1746-------> freq of popular category(family)
#### Name: Category, dtype: object


In [64]:
df['Category'].describe()

count       9360
unique        33
top       FAMILY
freq        1746
Name: Category, dtype: object

## "Analysis on multiple Columns"

##### dataset to array
##### Here we may deal with multiple different tyes of categoris we need the dataframe into category

In [79]:
#Here we may deal with multiple different tyes of categoris we need the dataframe into category
ar=df.values
ar

array([['Photo Editor & Candy Camera & Grid & ScrapBook',
        'ART_AND_DESIGN', 4.1, ..., 'January 7, 2018', '1.0.0',
        '4.0.3 and up'],
       ['Coloring book moana', 'ART_AND_DESIGN', 3.9, ...,
        'January 15, 2018', '2.0.0', '4.0.3 and up'],
       ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
        'ART_AND_DESIGN', 4.7, ..., 'August 1, 2018', '1.2.4',
        '4.0.3 and up'],
       ...,
       ['Fr. Mike Schmitz Audio Teachings', 'FAMILY', 5.0, ...,
        'July 6, 2018', '1.0', '4.1 and up'],
       ['The SCP Foundation DB fr nn5n', 'BOOKS_AND_REFERENCE', 4.5, ...,
        'January 19, 2015', 'Varies with device', 'Varies with device'],
       ['iHoroscope - 2018 Daily Horoscope & Astrology', 'LIFESTYLE',
        4.5, ..., 'July 25, 2018', 'Varies with device',
        'Varies with device']], dtype=object)

In [81]:
df.values.shape

(9360, 13)

### Q-how many  free apps are there in ART_AND_DESIGN

In [83]:
count = 0
for i in ar:
    if(i[1]=='ART_AND_DESIGN' and i[6]=='Free'):
        count+=1;
print("how many free apps are there in ART_AND_DESIGN: ",count)

how many free apps are there in ART_AND_DESIGN:  58


### Q-how many free apps are there in ART_AND_DESIGN

In [85]:
count = 0
for i in ar:
    if(i[1]=='ART_AND_DESIGN' and i[6]=='Paid'):
        count+=1;
print("how many Paid apps are there in ART_AND_DESIGN: ",count)

how many Paid apps are there in ART_AND_DESIGN:  3


### Q-how may apps are there in ART_AND_DESIGN with rating more than 4.5

In [94]:
count = 0
for i in ar:
    if(i[1]=='ART_AND_DESIGN' and i[2]>= 4.5):
        count+=1;
print("how may apps are there in ART_AND_DESIGN with rating more than 4.5: ",count)

how may apps are there in ART_AND_DESIGN with rating more than 4.5:  26


### Q-how may apps are there in family category with rating more than 4.5 and should be free

In [90]:
count = 0
for i in ar:
    if(i[1]=='FAMILY' and i[6]== 'Free' and i[2]>= 4.5):
        count+=1;
print("how may apps are there in family category with rating more than 4.5 and should be free: ",count)

how may apps are there in family category with rating more than 4.5 and should be free:  477


### Q-list all the free apps with rating more than 4.5 and should of family category

In [95]:
print("-"*150)
print("Q-list all the free apps with rating more than 4.5 and should of family category")
print("-"*150)
count = 0
for i in ar:
    if(i[1]=='FAMILY' and i[6]== 'Free' and i[2]>= 4.5):
        count+=1;
        print(i[0])

------------------------------------------------------------------------------------------------------------------------------------------------------
Q-list all the free apps with rating more than 4.5 and should of family category
------------------------------------------------------------------------------------------------------------------------------------------------------
YouTube Kids
ROBLOX
Mahjong
Super ABC! Learning games for kids! Preschool apps
Toy Pop Cubes
Candy Pop Story
Princess Coloring Book
Candy Smash
Happy Fruits Bomb - Cube Blast
Dog Run - Pet Dog Simulator
Puzzle Kids - Animals Shapes and Jigsaw Puzzles
Kids Educational :All in One
Toca Kitchen 2
No. Color - Color by Number, Number Coloring
Garden Fruit Legend
Candy Day
Learn To Draw Glow Princess
ABC Kids - Tracing & Phonics
Minion Rush: Despicable Me Official Game
Piano Kids - Music & Songs
Farming Simulator 14
Educational Games for Kids
No.Draw - Colors by Number 2018
Duolingo: Learn Languages Free
Hot Wheels:

# "Analysis of Dataset with Conditions"

In [98]:
#datarows belong to family in category column
df['Category']=="FAMILY"

0        False
1        False
2        False
3        False
4        False
         ...  
10834     True
10836     True
10837     True
10839    False
10840    False
Name: Category, Length: 9360, dtype: bool

### Q-records with are of Family application category in category column

In [101]:
df[df['Category']=="FAMILY"]

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
2014,YouTube Kids,FAMILY,4.5,470694,Varies with device,"50,000,000+",Free,0,Everyone,Entertainment;Music & Video,"August 3, 2018",3.43.3,4.1 and up
2015,Candy Bomb,FAMILY,4.4,42145,20M,"10,000,000+",Free,0,Everyone,Casual;Brain Games,"July 4, 2018",2.9.3181,4.0.3 and up
2016,ROBLOX,FAMILY,4.5,4449910,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
2017,Jewels Crush- Match 3 Puzzle,FAMILY,4.4,14774,19M,"1,000,000+",Free,0,Everyone,Casual;Brain Games,"July 23, 2018",1.9.3901,4.0.3 and up
2018,Coloring & Learn,FAMILY,4.4,12753,51M,"5,000,000+",Free,0,Everyone,Educational;Creativity,"July 17, 2018",1.49,4.0.3 and up
...,...,...,...,...,...,...,...,...,...,...,...,...,...
10820,Fr. Daoud Lamei,FAMILY,5.0,22,8.6M,"1,000+",Free,0,Teen,Education,"June 27, 2018",3.8.0,4.1 and up
10827,Fr Agnel Ambarnath,FAMILY,4.2,117,13M,"5,000+",Free,0,Everyone,Education,"June 13, 2018",2.0.20,4.0.3 and up
10834,FR Calculator,FAMILY,4.0,7,2.6M,500+,Free,0,Everyone,Education,"June 18, 2017",1.0.0,4.1 and up
10836,Sya9a Maroc - FR,FAMILY,4.5,38,53M,"5,000+",Free,0,Everyone,Education,"July 25, 2017",1.48,4.1 and up


In [ ]:
### Q-how may free apps are avaliable in ART_AND_DESIGN 

In [114]:
apps = df[df['Category']=="ART_AND_DESIGN"]
print(len(apps))
apps 

61


,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up
...,...,...,...,...,...,...,...,...,...,...,...,...,...
5258,AJ Styles Wallpaper 2018 - AJ Styles HD Wallpaper,ART_AND_DESIGN,4.0,18,3.6M,"1,000+",Free,0,Everyone,Art & Design,"March 22, 2018",1.1,4.0 and up
6930,Fantasy theme dark bw black building,ART_AND_DESIGN,4.8,41,1.9M,"5,000+",Free,0,Everyone,Art & Design,"March 20, 2018",1.0.2,4.0 and up
8679,Dp for girls,ART_AND_DESIGN,4.2,175,8.2M,"50,000+",Free,0,Everyone,Art & Design,"May 26, 2017",1.2,4.0.3 and up
8712,Name Art DP - Focus n Filter Text 2018,ART_AND_DESIGN,4.3,132,7.7M,"10,000+",Free,0,Everyone,Art & Design,"January 4, 2018",1,4.1 and up


In [112]:
freeapps = apps[apps['Type']=='Free']
print("total free ART_AND_DESIGN apps",len(freeapps))
freeapps

total free ART_AND_DESIGN apps 58


,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up
5,Paper flowers instructions,ART_AND_DESIGN,4.4,167,5.6M,"50,000+",Free,0,Everyone,Art & Design,"March 26, 2017",1.0,2.3 and up
6,Smoke Effect Photo Maker - Smoke Editor,ART_AND_DESIGN,3.8,178,19M,"50,000+",Free,0,Everyone,Art & Design,"April 26, 2018",1.1,4.0.3 and up
7,Infinite Painter,ART_AND_DESIGN,4.1,36815,29M,"1,000,000+",Free,0,Everyone,Art & Design,"June 14, 2018",6.1.61.1,4.2 and up
8,Garden Coloring Book,ART_AND_DESIGN,4.4,13791,33M,"1,000,000+",Free,0,Everyone,Art & Design,"September 20, 2017",2.9.2,3.0 and up
9,Kids Paint Free - Drawing Fun,ART_AND_DESIGN,4.7,121,3.1M,"10,000+",Free,0,Everyone,Art & Design;Creativity,"July 3, 2018",2.8,4.0.3 and up


In [ ]:
### Q-how may free apps are avaliable in ART_AND_DESIGN with rating more than 4.5

In [111]:
highratingapps = freeapps[freeapps['Rating']>4.5]
print("ART_AND_DESIGN with rating more than 4.5",highratingapps)
highratingapps

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
9,Kids Paint Free - Drawing Fun,ART_AND_DESIGN,4.7,121,3.1M,"10,000+",Free,0,Everyone,Art & Design;Creativity,"July 3, 2018",2.8,4.0.3 and up
13,Mandala Coloring Book,ART_AND_DESIGN,4.6,4326,21M,"100,000+",Free,0,Everyone,Art & Design,"June 26, 2018",1.0.4,4.4 and up
16,Photo Designer - Write your name with shapes,ART_AND_DESIGN,4.7,3632,5.5M,"500,000+",Free,0,Everyone,Art & Design,"July 31, 2018",3.1,4.1 and up
17,350 Diy Room Decor Ideas,ART_AND_DESIGN,4.5,27,17M,"10,000+",Free,0,Everyone,Art & Design,"November 7, 2017",1.0,2.3 and up
19,ibis Paint X,ART_AND_DESIGN,4.6,224399,31M,"10,000,000+",Free,0,Everyone,Art & Design,"July 30, 2018",5.5.4,4.1 and up
22,Superheroes Wallpapers | 4K Backgrounds,ART_AND_DESIGN,4.7,7699,4.2M,"500,000+",Free,0,Everyone 10+,Art & Design,"July 12, 2018",2.2.6.2,4.0.3 and up
24,HD Mickey Minnie Wallpapers,ART_AND_DESIGN,4.7,118,23M,"50,000+",Free,0,Everyone,Art & Design,"July 7, 2018",1.1.3,4.1 and up
25,Harley Quinn wallpapers HD,ART_AND_DESIGN,4.8,192,6.0M,"10,000+",Free,0,Everyone,Art & Design,"April 25, 2018",1.5,3.0 and up


# 'DataAnalysis using Grouping'

### Q-Name the Apps of ART_AND_DESIGN with rating more than 4.5 in ascending order with respective rating

In [119]:
highratingapps.sort_values(by = 'Rating', ascending= True)

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
17,350 Diy Room Decor Ideas,ART_AND_DESIGN,4.5,27,17M,"10,000+",Free,0,Everyone,Art & Design,"November 7, 2017",1.0,2.3 and up
41,Sad Poetry Photo Frames 2018,ART_AND_DESIGN,4.5,176,10M,"100,000+",Free,0,Everyone,Art & Design,"April 2, 2018",1.0,4.0.3 and up
32,Anime Manga Coloring Book,ART_AND_DESIGN,4.5,5035,11M,"100,000+",Free,0,Everyone,Art & Design,"July 19, 2018",2.20,4.0 and up
19,ibis Paint X,ART_AND_DESIGN,4.6,224399,31M,"10,000,000+",Free,0,Everyone,Art & Design,"July 30, 2018",5.5.4,4.1 and up
13,Mandala Coloring Book,ART_AND_DESIGN,4.6,4326,21M,"100,000+",Free,0,Everyone,Art & Design,"June 26, 2018",1.0.4,4.4 and up
46,Install images with music to make video withou...,ART_AND_DESIGN,4.6,1070,26M,"100,000+",Free,0,Everyone,Art & Design,"November 14, 2017",1.6,4.1 and up
9,Kids Paint Free - Drawing Fun,ART_AND_DESIGN,4.7,121,3.1M,"10,000+",Free,0,Everyone,Art & Design;Creativity,"July 3, 2018",2.8,4.0.3 and up
22,Superheroes Wallpapers | 4K Backgrounds,ART_AND_DESIGN,4.7,7699,4.2M,"500,000+",Free,0,Everyone 10+,Art & Design,"July 12, 2018",2.2.6.2,4.0.3 and up
16,Photo Designer - Write your name with shapes,ART_AND_DESIGN,4.7,3632,5.5M,"500,000+",Free,0,Everyone,Art & Design,"July 31, 2018",3.1,4.1 and up


In [120]:
#top 5 records
highratingapps.sort_values(by = 'Rating', ascending= True).head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
17,350 Diy Room Decor Ideas,ART_AND_DESIGN,4.5,27,17M,"10,000+",Free,0,Everyone,Art & Design,"November 7, 2017",1.0,2.3 and up
41,Sad Poetry Photo Frames 2018,ART_AND_DESIGN,4.5,176,10M,"100,000+",Free,0,Everyone,Art & Design,"April 2, 2018",1.0,4.0.3 and up
32,Anime Manga Coloring Book,ART_AND_DESIGN,4.5,5035,11M,"100,000+",Free,0,Everyone,Art & Design,"July 19, 2018",2.20,4.0 and up
19,ibis Paint X,ART_AND_DESIGN,4.6,224399,31M,"10,000,000+",Free,0,Everyone,Art & Design,"July 30, 2018",5.5.4,4.1 and up


### Q-Name the Apps of ART_AND_DESIGN with rating more than 4.5 in descending order with respective rating

In [121]:
highratingapps.sort_values(by = 'Rating', ascending= False).head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
8888,Spring flowers theme couleurs d t space,ART_AND_DESIGN,5.0,1,2.9M,100+,Free,0,Everyone,Art & Design,"April 18, 2018",1.0.2,4.0 and up
25,Harley Quinn wallpapers HD,ART_AND_DESIGN,4.8,192,6.0M,"10,000+",Free,0,Everyone,Art & Design,"April 25, 2018",1.5,3.0 and up
3982,Cardi B Wallpaper,ART_AND_DESIGN,4.8,253,3.7M,"50,000+",Free,0,Everyone,Art & Design,"November 1, 2017",1.0.0,4.0 and up
5254,AJ Styles HD Wallpapers,ART_AND_DESIGN,4.8,48,25M,"5,000+",Free,0,Everyone,Art & Design,"August 4, 2018",1.1.1,4.1 and up
6930,Fantasy theme dark bw black building,ART_AND_DESIGN,4.8,41,1.9M,"5,000+",Free,0,Everyone,Art & Design,"March 20, 2018",1.0.2,4.0 and up


### Q-which category has the maximun rating

In [128]:
df.groupby('Category').mean(numeric_only=True)['Rating'].sort_values(ascending=False)

Category
EVENTS                 4.435556
EDUCATION              4.389032
ART_AND_DESIGN         4.377049
BOOKS_AND_REFERENCE    4.346067
PERSONALIZATION        4.335256
PARENTING              4.300000
GAME                   4.286326
BEAUTY                 4.278571
HEALTH_AND_FITNESS     4.277104
SHOPPING               4.259664
SOCIAL                 4.255598
WEATHER                4.244000
SPORTS                 4.223511
PRODUCTIVITY           4.211396
HOUSE_AND_HOME         4.197368
FAMILY                 4.192383
PHOTOGRAPHY            4.192114
AUTO_AND_VEHICLES      4.190411
MEDICAL                4.189143
LIBRARIES_AND_DEMO     4.179688
FOOD_AND_DRINK         4.166972
COMMUNICATION          4.158537
COMICS                 4.155172
NEWS_AND_MAGAZINES     4.132189
FINANCE                4.131889
ENTERTAINMENT          4.126174
BUSINESS               4.121452
TRAVEL_AND_LOCAL       4.109292
LIFESTYLE              4.094904
VIDEO_PLAYERS          4.063750
MAPS_AND_NAVIGATION    4.051613